docs/examples/quickstart/test_quickstarts.rb

```diff
@@ -0,0 +1,143 @@
+# frozen_string_literal: true
+
+require 'English'
+require 'fileutils'
+require 'timeout'
+
+require 'bundler'
+
+CURL_ERROR = "curl: (7) Failed to connect to localhost port 5050: Connection refused\n"
+CURL_EMPTY = "curl: (52) Empty reply from server\n"
+
+quickstarts = [
+  { lang: 'java', setup: 'make build', server: 'make run' },
+  { lang: 'nodejs', setup: 'npm i', server: 'npm start' },
+  { lang: 'python', setup: 'pip install --upgrade -r requirements.txt', server: 'python server.py' },
+  { lang: 'ruby', setup: 'bundle', server: 'bundle exec ruby server.rb' },
+  { lang: 'rust', setup: 'cargo build', server: 'cargo run' }
+]
+
+def start_server(server, user, expense_id)
+  server = spawn server
+  received = CURL_ERROR
+  while [CURL_ERROR, CURL_EMPTY].include? received
+    sleep 0.5
+    received = `curl -sSH "user: #{user}" localhost:5050/expenses/#{expense_id} 2>&1`
+  end
+  [server, received]
+end
+
+def ensure_port_5050_is_open
+  until (server = `lsof -ti :5050 2>&1`.split.first.to_i).zero?
+    sleep 0.5
+    Process.kill 'TERM', server
+  end
+rescue Errno::ESRCH => e
+  puts "#{e}: #{server}"
+end
+
+def kill_server(server)
+  return if server.nil?
+
+  Process.kill 'TERM', server
+  Process.wait2 server
+rescue Errno::ESRCH => e
+  puts "#{e}: #{server}"
+end
+
+# rubocop:disable Metrics/BlockLength
+
+quickstarts.each do |qs|
+  lang = qs[:lang]
+  qs_dir = "oso-#{lang}-quickstart"
+  Bundler.with_unbundled_env do
```

Comment:
This is a workaround for the fact that we're using Ruby to orchestrate these tests and one of the examples under test is a Ruby project. Without this, building and running the Ruby quickstart guide would use an unclean environment, e.g., would use the local version of the `oso-oso` gem (in the `languages/ruby` path) instead of the latest release.

Reply:
Thanks.